packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js

```diff
@@ -0,0 +1,189 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { parseFixed } = require("@ethersproject/bignumber");
+
+// An implementation of PriceFeedInterface that takes as input two sets ("baskets") of price feeds,
+// computes the average price feed for each basket, and returns the spread between the two averages.
+// !!Note: This PriceFeed assumes that the baselinePriceFeeds, experimentalPriceFeed, and denominatorPriceFeed
+// are all returning prices in the same precision as `decimals`.
+class BasketSpreadPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs new BasketSpreadPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {List} baselinePriceFeeds The baseline list of priceFeeds to compute the average of. All elements must be of type PriceFeedInterface.
+   *      Must be an array of at least one element.
+   * @param {List} experimentalPriceFeed The baseline list of priceFeeds to compute the average of. All elements must be of type PriceFeedInterface.
+   *      Must be an array of at least one element.
+   * @param {Object} denominatorPriceFeed We divide the price spread between the baseline and experimental baskets by this denominator price
+   *      in order to "denominate" the basket spread price in a specified unit. For example, we might want to express the basekt spread in terms
+   *      of ETH-USD.
+   */
+  constructor(web3, logger, baselinePriceFeeds, experimentalPriceFeeds, denominatorPriceFeed) {
+    super();
+
+    if (baselinePriceFeeds.length === 0 || experimentalPriceFeeds.length === 0) {
+      throw new Error("BasketSpreadPriceFeed cannot be constructed with empty baseline or experimental baskets.");
+    }
+
+    this.baselinePriceFeeds = baselinePriceFeeds;
+    this.experimentalPriceFeeds = experimentalPriceFeeds;
+    this.denominatorPriceFeed = denominatorPriceFeed;
+
+    // Helper modules.
+    this.web3 = web3;
+    this.toBN = this.web3.utils.toBN;
+    this.toWei = this.web3.utils.toWei;
+    this.logger = logger;
+
+    // The precision that the user wants to return prices in is determined by the denominator price feed.
+    this.decimals = denominatorPriceFeed.getPriceFeedDecimals();
+
+    // Scale `number` by 10**decimals.
+    this.convertDecimals = number => {
```

Comment:
small request: can you please name this `convertPriceFeedDecimals` so that it's _Really_ clear that this is scaling by the number of `decimals`?

Reply:
Done